pkgs/by-name/up/uppaal/package.nix

```
@@ -0,0 +1,88 @@
+{
+  lib,
+  stdenv,
+  fetchzip,
+  makeWrapper,
+  jdk17,
+  makeDesktopItem,
+  copyDesktopItems,
+  wrapGAppsHook,
+  shared-mime-info,
+  gdk-pixbuf,
+}:
+
+stdenv.mkDerivation rec {
+  pname = "uppaal";
+  version = "5.0";
+  subversion = "0";
+  platform = "linux64";
+
+  src = fetchzip {
+    url = "https://download.${pname}.org/${pname}-${version}/${pname}-${version}.${subversion}/${pname}-${version}.${subversion}-${platform}.zip";
```

While I get the feeling of wanting to make things as dynamic as possible, try to avoid using `pname` for things that happen to be the same literal as the package name but isn't logically related to it.

```suggestion
    url = "https://download.uppaal.org/uppaal-${version}/uppaal-${version}.${subversion}/uppaal-${version}.${subversion}-${platform}.zip";
```